dangerzone/isolation_provider/base.py

```diff
@@ -86,11 +88,17 @@ class IsolationProvider(ABC):
     Abstracts an isolation provider
     """
 
-    def __init__(self) -> None:
-        if getattr(sys, "dangerzone_dev", False) is True:
+    def __init__(self, debug: bool = False) -> None:
+        self.debug = debug
+        if self.should_capture_stderr():
             self.proc_stderr = subprocess.PIPE
         else:
             self.proc_stderr = subprocess.DEVNULL
+        # Store the proc stderr in memory
+        self.stderr = BytesIO()
```

Comment:
Kind of a minor note for now, but will be important in the future: I'd suggest not storing objects in the `IsolationProvider` instance that are affected by the conversion. It's not important for now, but when we will have multiple conversions in flight, then it won't work.

The `BytesIO` object in particular can be moved to the `doc_to_pixels_proc()` context manager, and get passed to the stderr thread.

Reply:
Thanks for mentioning this. I even think that we could use composition rather than inheritance in the future, to make that impossible :-)

I did the changes you mentioned in 3b961e6b9009779cd56609459eda84c1bb56469f